stdlib/src/utils/span.mojo

```diff
@@ -241,3 +241,17 @@ struct Span[
         debug_assert(len(self) == len(other), "Spans must be of equal length")
         for i in range(len(self)):
             self[i] = other[i]
+
+    @always_inline
+    fn fill[lifetime: MutableLifetime](self: Span[T, lifetime], value: T):
```

Comment:
**Question** Does this method still compile if written as `inout self`, without the explicit lifetime parameter?

If so, I suggest we do that to keep the signature simple in cases where that's possible.

Reply:
It doesn't work. I think the nuance here is that `inout` refers to immutable of the container, but not the data that it points to. So:
```
fn subslice[lifetime: ImmutableLifetime](inout self: Span[T, lifetime], value: T):
    self = self[0:3]
```
works just fine, since it makes no assumptions about the mutability of the data itself, just the value stored in the container type.

Reply:
Ah, that makes perfect sense, thanks for the explanation. This was just me misreading / misunderstanding the signature meaning—sorry for the noise.